Pegasus/include/Geometry.hpp

```diff
@@ -354,23 +312,78 @@ struct IntersectionCache<Box, Box> : IntersectionCacheBase
     double penetration = 0;
 };
 
+/** Data structure to store ray factors for Ray, Sphere collisions */
+struct RayIntersectionFactors
+{
+    double tMin;
+    double tMax;
+};
+
 /**
- * @brief Performs initial calculations and writes it into the cache object
- * 
- * Required to be called before any other proximity query function, otherwise results are undefined
- * @tparam ShapeA SimpleShape or SimpleShape derived object
- * @tparam ShapeB SimpleShape or SimpleShape derived object
- * @param[in] a pointer to the ShapeA type object
- * @param[in] b pointer to the ShapeB type object
- * @param[out] cacheBase pointer to the IntersectionCacheBase or IntersectionCacheBase derived object
+* @brief Calculates wether a ray and a sphere are intersecting
```

Comment:
`whether`

Also what does it `calculate`? Since it returns `bool` and has no `@param[out]` it simply checks whether they are intersecting or not